File: js/samples/04.ai.b.messageExtensions.gptME/src/cards/editView.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { Attachment, CardFactory } from '@microsoft/teams-core';

/**
 * @param post
 * @param previewMode
 */
export function createEditView(post: string, previewMode: boolean): Attachment {
    return CardFactory.adaptiveCard({
        $schema: 'http://adaptivecards.io/schemas/adaptive-card.json',
        type: 'AdaptiveCard',
        version: '1.4',
        body: [
            {
                type: 'Input.Text',
                id: 'prompt',
                placeholder: 'Enter a new prompt that updates the post below',
                isMultiline: true
            },
            {
                type: 'Container',
                minHeight: '160px',
                verticalContentAlignment: 'Center',
                items: [
                    {
                        type: 'TextBlock',
                        wrap: true,
                        text: post
                    },
                    {
                        type: 'Input.Text',
                        id: 'post',
                        isVisible: false,
                        value: post
                    }
                ]
            }
        ],
        actions: [
            {
                type: 'Action.Submit',
                title: 'Update',
                data: {
                    verb: 'update'
                }
            },
            {
                type: 'Action.Submit',
                title: previewMode ? 'Preview' : 'Post',
                data: {
                    verb: previewMode ? 'preview' : 'post'
                }
            }
        ]
    });
}

File: python/samples/04.ai.b.messageExtensions.AI-ME/src/cards/edit_view.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

from botbuilder.core import CardFactory
from botbuilder.schema import Attachment


def create_edit_view(post, preview_mode) -> Attachment:
    return CardFactory.adaptive_card(
        {
            "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
            "type": "AdaptiveCard",
            "version": "1.4",
            "body": [
                {
                    "type": "Input.Text",
                    "id": "prompt",
                    "placeholder": "Enter a new prompt that updates the post below",
                    "isMultiline": True,
                },
                {
                    "type": "Container",
                    "minHeight": "160px",
                    "verticalContentAlignment": "Center",
                    "items": [
                        {"type": "TextBlock", "wrap": True, "text": post},
                        {"type": "Input.Text", "id": "post", "isVisible": False, "value": post},
                    ],
                },
            ],
            "actions": [
                {"type": "Action.Submit", "title": "Update", "data": {"verb": "update"}},
                {
                    "type": "Action.Submit",
                    "title": "Preview" if preview_mode else "Post",
                    "data": {"verb": "preview" if preview_mode else "post"},
                },
            ],
        }
    )
